src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -184,6 +184,8 @@ struct processing_module {
 	struct output_stream_buffer *output_buffers;
 	uint32_t num_input_buffers; /**< number of input buffers */
 	uint32_t num_output_buffers; /**< number of output buffers */
+	struct comp_buffer *source; /**< single source buffer for multi_source_or_sink = false */
```

Comment:
Please don't call it sink/source - may be confusions with struct sof_source / struct sof_sink interface

`struct comp_buffer source_comp_buffer;` 

or simiar